# *Apodemia mormo s.l.* GBS paper

Authors: J.C. Oliver, J.R. Dupuis, T. Longcore, J. Johnson, F.A.H. Sperling
Additional authors: Bryan Brunet?

## Abstract

## Introduction
Genetic markers have long been looked to for setting management policies, largely with the intent of applying objective criteria to identify taxa of conservation priority (REFS in [Allendorf 2016](#allendorf-2016)). Genetic differentiation or genetic 'uniqueness' has been used with a variety of descriptions, including Evolutionary Significant Units (REF), Distinct Population Segments (REF), and Management Units [Moritz 1994](#moritz-1994). Criteria for defining populations warranting conservation action include reciprocal monophyly (REF) and significant differences in allele frequencies, applying one of a number of tests for genetic differentiation (REF). Such approaches have mixed results, with some genetic analyses corroborating other evidence supporting the distinctness of a taxon (REF), some suggesting no genetic differentiation of populations already under conservation focus (REF), and some identifying previously unrecognized genetic lineages ('cryptic diversity') (REF).

Many of the early challenges to the use of genetic markers in prioritizing conservation efforts were due to a relatively small sampling of genetic markers. Indeed, many studies relied on mitochondrial DNA (mtDNA) (REF), which has the benefit of a relatively high substitution rate, but is a single, non-recombining marker inherited from a single parent. Advances in genetic assays, especially next-generation sequencing technologies, have provided a means to overcome the problems associated with assays based on a trivial portion of the genome ([Allendorf 2016](#allendorf-2016)). Increasing the number of markers by orders of magnitude has made it less likely to miss important genetic variation due to insufficient sampling (REF). Furthermore, deeper coverage of genomes affords insights into previously undetectable selective pressures (REF) and demographic processes (REF). Here we apply next-generation sequencing to an endangered insect, in an effort to provide information useful to conservation priorities. Specifically we assess the support for genetic differentiation and the relationship to other populations and taxa.

The Mormon Metalmark species complex includes _Apodemia mormo_ (Felder & Felder), _A. virgulti_ (Behr), _A. mejicanus_ (Behr) (Lepidoptera: Riodinidae)...This complex is distributed across western North America and is characterized by considerable among-population variation in caterpillar host-plant use, phenology, and wing pattern morphology (Scott 1986; Pratt & Ballmer 1991, Opler & Wright? More?). This variation has undoubtedly contributed to a proliferation of a number of taxonomic entities being described: there are at least 17 currently subspecies in the complex (Pelham's Catalog), some of which occur syntopically (Pratt & Ballmer 1991). In some cases, this variation is accompanied by genetic differentiation, in some cases over relatively small geographic scales ([Crawford et al. 2011](#crawford-et-al-2011), [Proshek et al. 2013](#proshek-et-al-2013)), and a number of populations in this species complex are under or are being considered for conservation protection ([Crawford et al. 2011](#crawford-et-al-2011), [Proshek et al. 2013](#proshek-et-al-2013), [Proshek et al. 2015](#proshek-et-al-2015)).

The focus of this work is the endangered subspecies, Lange's metalmark, _Apodemia mormo langei_,

1. Bigger question: Genetics in ESU (DPS: Distinct Population Segments)
    + See also Management Units of Mortiz 1994
2. Next-gen in ESU, see commentary by Allendorf Mol. Ecol. 2016, no longer single or handful of markers, which could miss genetic variation... (see Toews & Brelsford 2012 referenced in Allendorf 2016)
3. Introduce _A. mormo_. What do we know about population subdivision in _A. mormo_, touch on taxonomic mess that is _A. mormo s.l._
4. Focus on _A. mormo langei_.
    + Some basic biology
    + What questions did we ask?
    + What methods did we use?

<a name="allendorf-2016">Allendorf, FW</a> 2016. Mol Ecol [doi: 10.1111/mec.13948](http://dx.doi.org/10.1111/mec.13948)

<a name="crawford-et-al-2011">Crawford et al.</a> 2011. Conserv Genet [doi: 10.1007/s10592-011-0202-4](http://dx.doi.org/10.1007/s10592-011-0202-4)

<a name="moritz-1994">Moritz C</a> 1994. Trends Ecol Evol [doi: 10.1016/0169-5347(94)90057-4](http://dx.doi.org/10.1016/0169-5347%2894%2990057-4)

<a name="proshek-et-al-2013">Proshek et al.</a> 2013. J Insect Conserv [doi: 10.1007/s10841-012-9494-z](http://dx.doi.org/10.1007/s10841-012-9494-z)

<a name="proshek-et-al-2015">Proshek et al.</a> 2015. BMC Evol Biol [doi: 10.1186/s12862-015-0354-9](http://dx.doi.org/10.1186/s12862-015-0354-9)