## Reviews and commentary

Comments of the editor and reviewers are in _italics_; my responses follow them in plain text.

### Editor
_As you can see the two reviewers mainly took you up on the general conclusions, in particular the 'subjectivity' issue. I have to agree that much of this is problematic for the reasons stated by the reviewers and, while stimulating for a general discussion, it isn't pertinent to the current study._

Hmmm...I still think it is, but do we need to make a stronger case? After all, studies such as this have the potential to influence policy.

_As one of the reviewer states, a lot of this could have been written a long time ago in pre-genomics times. I think your emphasis should be on the data themselves._

Similar things _could_ have been written, but all these data were supposed to save us the trouble of making subjective decisions. Advocates of "data-driven" prioritization would argue _langei_ warrants no more protection than any other population of _A. mormo_.

_As a phylogeneticist, I am intrigued by your cloudogram but wasn't given much information what this is based on.  For example, how many trees and characters each have you used?_

We can add these details.

_Can these short RAD markers provide robust trees?_

Cite [Cariou et al. (2013, _Ecol Evol_)](https://dx.doi.org/10.1002/ece3.512) as _in silico_ support and [Herrera & Shank (2016, _Mol Phylogenet Evol_)](https://dx.doi.org/10.1016/j.ympev.2016.03.010) for a worked example.

_What would happen if you used the other markers with more missing data but more individuals?  Likewise I would have liked a discussion about the power of these markers for population genetics (and the detailed analysis which seems very well done), relative to the COI sequences and microsatellite
data already in hand._

This seems outside the scope of the current study. We're not out to necessarily compare and contrast different markers, although we do actually do that in the discussion to some degree.

_In fact I found myself largely in the dark how much already is known about this species. From a quick look at the 2015 paper, it seems quite a lot! In which case that should be the starting point, not the general question about ESUs. The main question to me is, what have we learned from using the genomics approaches we didn't already know from conventional markers?_

While I think we still have reason to advocate for the ESU discussion, I could see adding more background to the Introduction to set the stage.

### Reviewer 1
_This paper brings to bear considerable new data and strong analytical prowess to a series of important questions with general relevance, but would be helped enormously by the authors' limiting themselves to fewer big-ticket items on the vast conceptual smorgasbord before them, and by tightening up the language a little. I think this can be a very good paper and although I think it requires some revision, most of it can be confined to the text, and I hope my review will be seen as constructive._

_In this paper, the authors tackle the inevitable outcome of NGS data that many more distinct "units" can be revealed than with the old onegene-twogene-threegene approach and, depending on the taxa, possibly on par with available infraspecific names (if it were possible to sequence the types). Here we have a case where a named subspecies turns out to be no less distinct than any one of a number of other isolated populations or metapopulations, the authors' asking the question of "whether Lange's metalmark qualifies as an ESU."_

Not sure these two introductory paragraphs provide any actionable feedback...

_As I see it, these potentially exciting new data that could not have been generated, much less analyzed, 20 years ago are brought to bear on a deceptively straightforward question of where to draw the line at "evolutionary significance." This question touches on everything from the mechanics of listing under the ESA and the purview of taxonomic nomenclature (and how that impacts the same mechanics of listing under the ESA...) to population-level management, i.e. actual, real-time conservation. That last of which is given fairly short shrift except in a somewhat removed sense of prioritization, which has negative connotations of quotas and zero-sum games (as in, "OK, we get to list five insects this year...what'll they be?). Unlike the data and analyses, though, there may be little in the discussion of these matters that could not have been written 20 or 30 years ago. Part of the reason for this is a somewhat thin treatment of the literature on certain topics. Notwithstanding
the tautological use of "unit", for example, it is not sufficient to cite 3 papers with an average age of 24 years to support the statement that "conservation biologists often use the Evolutionarily Significant Unit (ESU) to address conservation units at this [population-level] scale."_

OK...we could cite more recent papers, such as [Neaves et al. (2016, _PLoS ONE_)](https://dx.doi.org/10.1371/journal.pone.0162207) and [Stiefel et al. (2015, _Idaho Natural Production Monitoring and Evaluation, 2014 annual report_)](https://collaboration.idfg.idaho.gov/FisheriesTechnicalReports/Res15-14Stiefel2014%20Natural%20Production%20Monitoring%20and%20Evaluation.pdf).

_More generally, I feel the paper could be improved by narrowing the range of target topics that are dealt glancing blows but not treated in depth. By aiming too broad or too high, the text fails in places to clarify where and how the data speak directly to a range of questions that still may be beyond their power to treat effectively. The prose surrounding "subjectivity management" (although I love the phrase) and the mechanics of federal law may be read as though the authors are outside their comfort zone._

OK, so what should we focus on? I can see the reviewer's point to a degree.

_Backing up a bit, the semantic morass of the "evolutionary significant unit" (ESU) was an attempt to incorporate biological reality and philosophical desirability with legal criteria for undertaking conservation. Practically speaking, the endeavor to identify and legitimize ESU's rests within that aspect of conservation focused on protecting organisms in unitary terms, be they entities with formal taxonomic epithets or otherwise; in other words, listing. The problem was that articulating both the philosophical and the legal issues relied on vague or weakly articulated characterizations of the biological ones and, worse, that they did so in conflicting ways. Every letter of the ESU acronym has been debated and argued over to little avail, most of the combatants retreating to claims of "flexibility" either to deflect or generate accusations of "subjectivity" (which is the same thing). The biological issues range from circumscribing species or units operationally to choosing surrogate metrics for distinctness and inferring adaptation from genomic data. I feel one specific cohort of the literature was overlooked, namely the various approaches to circumscribing species based on genetic data and in particular the bearing of diagnosability on ESUs. Since I've obviously made known my thinking concerning the importance of diagnostics, I don't wish to appear biased or desirous of agreement or citation, but as someone familiar with published discussions, I found the treatment lacking. This may appear more conspicuous given the paper's emphasis on "distinctiveness." There appear to be instances where "evolution" is conflated with diagnosibility (also concepts/definitions with criteria), but beyond that I tend to think of the issue as more nuanced than a binary debate between operationalism and subjectivity, for example. Quite an array of relatively recent approaches have been brought to bear on species delimitation, within and outside the conservation literature. Some are at odds, but all claim rigor. When my co-authors and I first (and no doubt imperfectly) examined some of the criteria for recognizing conservation units and their bearing on some of the issues now raised by Dupuis et al., it was before approaches such as DNA taxonomy and generalized Yule models were being discussed, and some of us probably wouldn't have agreed with them anyway. But the point is that they exist and may have bearing. And so I do suggest taking a look at some of that literature, warts and all. Some references are included below._

Maybe we describe, early on, the scope of what we are going after?

_The real elephant in the room is not just the data that arbitrate how distinctive populations are, but which of those data achieve the requisite evolutionary significance to warrant the designation as an ESU. It is not a new observation that using poorly defined words interchangeably with imprecise or unquantifiable metrics merely amplifies a conceptual problem for legislative, philosophical, and most importantly practical purposes. As it stands, this paper touches on all three of these, but shies away from really nailing down any of them decisively. From a practical standpoint, the onus of proving that any components of a would-be ESU are "adaptive" has always been ridiculous (not merely "challenging"). But I feel as though a mixed message is being sent: The authors' stated intention is to "assess their support for genomic distinctiveness and evolutionary significance". They state "Here, we present the first genome-wide assessment of evolutionary significance in the Mormon
metalmark" and "With this large dataset, we assess support for genomic distinctiveness and evolutionary significance of A. m. langei..." But as early as the Abstract we're told "Our data are not sufficient to assess whether this genetic variation is adaptive, which presents a challenge for characterizing this endangered subspecies as an ESU," and later that "there is little support for recognizing it as an ESU based on current definitions."_

So we back off the whole "adaptive-significance" thing? Mention it in the discussion? I think it is still important to report negative results.

_If indeed there is "little support" for the recognition of A. mormo as an ESU given the amount of data generated, there is as little realistic a prospect of finding such support in the near future, due to (1) the highly variable and complex nature of the A. mormo species group, and (2) the general lack of detailed ecological knowledge that could be used to assess the group's adaptive genetic variation. Part of the frustration may derive from a desire to reconcile nomenclature with biology, and if so this is a pitfall wisely avoided. Except possibly among amateur butterfly taxonomists, entomologists do not commonly describe subspecies, much less advocate subspecific names as representative of biological entities. One could even argue that intraspecific nomenclature is a catch-can for systematizing a range of population-level variation that helps, at most, communicate conservation needs. I would suggest minimizing discussion of whether subspecific names correspond to biological realities, and bottle any surprise when they do not._

Bold. Nomenclature still has policy implications, regardless of one's opinions of subspecies. See chapter 3 of [NRC (1995, _Science and the Endangered Species Act_)](https://www.ncbi.nlm.nih.gov/books/NBK232366/).

_Regardless of whether species-based conservation has any hope of achieving any kind of sustainable management at the landscape level, it's probably not a bad idea to state outright and up front that this is the arena to which the paper's confined. As a supporter of species-based conservation as a component of a larger approach, I find this study may be, if anything, an outstanding example of the potential rabbit holes we risk going down when conservation is dominated or overparameterized by outdated endangered species legislation._

I pretty much agree and think we have the opportunity to focus on the need for common sense in policy decisions (gasp).

### Reviewer 2:
_The authors use a case study of metalmark butterflies to tackle a big issue/ subject of debate in conservation genetics on how to diagnose units for conservation. The empirical work is solid and provides a nice foundation for their arguments. This is an important contribution. I think the points that the authors make are good ones that will serve to foster more discussion._

So far so good.

_One counter argument that could be made comes from the way in which the authors frame the issue. The arguments about adaptive genetic variation in the ESU definition come from a long tradition of prioritizing conservation of evolutionary potential and the subsidiary issue of minimizing outbreeding depression in management plans. The authors argue (e.g. line 423) that it "is questionable whether these large [population genomics] datasets bring conservation biologists any closer to identifying adaptive genetic variation that can objectively define ESUs", specifically with respect to adaptive differentiation. The authors are quite correct. However, I would argue that basic population genetics data of any sort (microsatellites, mtDNA or SNPs) never provided information on adaptive differentiation. Since the introduction of ESU ideas, and especially since Crandall et al. 2000, **it has been clear that surveying geographic patterns of genetic variation has almost no utility for the adaptive part of the ESU definition**._ [emphasis added]

Is that last part right? Has _no one_ been advocating for genome scans of adaptive significance? Depending on how much effort we put into the adaptive significance angle, I think we should be able to provide citations illustrating that there _is_ still interest in this type of approach. Maybe [Larson et al. (2017, _Can J Fish Aquatic Sci_)](https://dx.doi.org/10.1139/cjfas-2017-0116)?

_The authors mention that genome scans (or other "outlier" analyses) could help, but they rightly point out that this is a weak approach and problematic without a clear understanding of determinants of fitness. Of course, the only defensible way to get at this aspect is through experimentation - reciprocal transplant experiments or experimental evolution - which is almost never feasible given the constraints of small populations and the ethical considerations of conducting such experiments with taxa of conservation concern. I would have liked the authors to clearly confront this point: population genetics surveys by themselves can never provide a full diagnosis of ESUs._

Hrm..."never"? I think that _a priori_ we cannot expect genetic data to diagnose ESUs, but _a posteriori_, genetic data can do a pretty good job of delimiting units in _some_ cases. I don't agree with the reviewer's absolutism.

_I also take some issue with the authors discussion of "subjective criteria" in defining ESUs. Yes, their is subjectivity in conservation biology. There is subjectivity in the easier part of the ESU definition, distinctiveness - how much differentiation is sufficient? But I think this is due to the practice of conservation genetics and it need not be this way. We could for example declare thresholds._

Did someone say barcode? Arbitrary thresholds seem like subjective criteria in an objectivity cloak.

_We could also come to consensus about adaptive differentiation (assuming we could do the required experiments). Thus, the subjectivity arguments seem to me to be a separate issue from the authors main points._

Damn. I think we missed something, because in my opinion the subjectivity issue is a pretty important point.

_line 141: as an aside and a question out of sheer curiuosity, does adding the C to the primer actually result in complexity reduction? It has been my experience that most of the high fidelity polymerases used for population genomics library preparation probably simply "repair" this mismatch leading to no complexity reduction._

_line 142: the is not clear. If another 96-plex library that has nothing to do with this project was prepared, why mention it? Unless the libraries from different taxa were mixed - ?_

_line 153: I do not understand "assemble a catalog de novo"._

Your bailiwick, JD.

_line 231: It might be noted that Legendre et al. (2015) (among other publications) have pointed out that Mantel tests can be problematic for sampling designs that are not geographically regular and recommend distance-based Moran's eigenvector mapping (dbMEM) as a robust alternative._

Well that sounds like fun. Let me do some research.

_Supplement 2 figures: for the barplots, what are major and minor modes. I am not familiar with this and none of this seems to be mentioned in the methods. Can this be interpreted to mean that structure runs did not converge? If so, then is this more about model fit than descriptions of the data?_

Good question. We could add something about convergence to the methods/results. It probably wouldn't hurt to update the figure (S3), replacing "N cluster" and "S cluster" with "North cluster" and "South cluster", respectively (this is explained in the legend, but it seems like an easy fix).